base/intfuncs.jl

```diff
@@ -387,6 +354,10 @@ julia> nextpow(4, 16)
 See also [`prevpow`](@ref).
 """
 function nextpow(a::Real, x::Real)
+    # Special case fast path for Integer/Unsiged a == 2.
+    # This is a very common case. Constant prop will make sure that a call site
+    # specified as `nextpow(2, x)` will get this special case inlined.
+    a == 2 && (isa(x, Integer) || isa(x, Unsigned)) && return _nextpow2(x)
```

Comment:
`Unsigned <: Integer`, so isn't this equivalent to just checking `isa(x, Integer)`?